cpp/cmake/thirdparty/get_faiss.cmake

```diff
@@ -24,22 +24,34 @@ function(find_and_configure_faiss)
         LIBRARY_NAMES faiss
     )
 
+    set(BUILD_SHARED_LIBS OFF)
+    if (NOT BUILD_STATIC_LIBS)
+        set(BUILD_SHARED_LIBS ON)
+    endif()
+
     rapids_cpm_find(FAISS ${PKG_VERSION}
-        GLOBAL_TARGETS  faiss
+        GLOBAL_TARGETS   faiss FAISS::FAISS
+        BUILD_EXPORT_SET raft-exports
         CPM_ARGS
           GIT_REPOSITORY  https://github.com/facebookresearch/faiss.git
           GIT_TAG         ${PKG_PINNED_TAG}
           OPTIONS
             "FAISS_ENABLE_PYTHON OFF"
-            "BUILD_SHARED_LIBS OFF"
+            "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
             "CUDAToolkit_ROOT ${CUDAToolkit_LIBRARY_DIR}"
             "FAISS_ENABLE_GPU ON"
             "BUILD_TESTING OFF"
             "CMAKE_MESSAGE_LOG_LEVEL VERBOSE"
     )
 
     if(FAISS_ADDED)
-      set(FAISS_GPU_HEADERS ${FAISS_SOURCE_DIR} PARENT_SCOPE)
+        target_include_directories(faiss INTERFACE $<BUILD_INTERFACE:${FAISS_SOURCE_DIR}>)
+        install(TARGETS faiss EXPORT faiss-exports)
+        rapids_export(BUILD faiss
+            EXPORT_SET faiss-exports
+            GLOBAL_TARGETS faiss
+            NAMESPACE raft::
+            LANGUAGES CUDA)
     endif()
 
```

Comment:
We need to add the following:

```cmake
    # We need to tell cmake where it can find the generated faiss-config.cmake we wrote.
    
    include("${rapids-cmake-dir}/export/find_package_root.cmake")
    rapids_export_find_package_root(BUILD faiss [=[${CMAKE_CURRENT_LIST_DIR}]=] raft-exports)
```
https://docs.rapids.ai/api/rapids-cmake/nightly/command/rapids_export_find_package_root.html

Reply:
Will also need the same for INSTALL